### PR TITLE
To/FromJSONKey instances for Credential and Voter 

### LINKED ISSUE #3580: Create a user friendly text representation for Credential and Voter
This is needed for ToJSON/ToJSONKey instances and potentially for other uses cases, such as error reporting.

In order to use a type as a Key in JSON it should be represented as a flat text.

### PR DESCRIPTION
**Pull request by teodanciu**
# Description

Closes https://github.com/input-output-hk/cardano-ledger/issues/3580

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [ ] Self-reviewed the diff

### CODE REVIEW

File: eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Procedures.hs [hunk 1]
```diff
@@ -162,9 +168,19 @@ data Voter c
...
 instance Crypto c => ToJSON (Voter c)
 
--- TODO: Make it nicer, eg. "DRep-ScriptHash<0xdeadbeef>..."
--- Will also need a ToJSONKey instance for Credential.
-instance Crypto c => ToJSONKey (Voter c)
+instance Crypto c => ToJSONKey (Voter c) where
+  toJSONKey = toJSONKeyText prefixVoter
+    where
+      prefixVoter (CommitteeVoter (KeyHashObj (KeyHash kh))) =
+        "committee-keyhash-" <> hashToTextAsHex kh
```
Comment (teodanciu) [inline eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Procedures.hs]: I can't tell if it's possible here to reuse the `ToJSONKey` instance of the credential, which is what I would like to do.
`toJSONKey` returns `ToJSONKeyFunctions` without taking any arguments, so not sure how I would apply this function to the Credential :thinking:

Comment (lehins) [reply eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Procedures.hs]: Yes you can create a separate function `credentialToText` and reuse it here and in the ToJSONKey instance for Credential

Comment (lehins) [inline eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Procedures.hs]: Like so. Also a nice approach to reduce amount of code on the screen with LambdaCase:
```suggestion
  toJSONKey = toJSONKeyText $ \case
    CommitteeVoter cred -> "committee-" <> credentialToText cred
```

Comment (teodanciu) [reply eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Procedures.hs]: that's a good idea. thanks

**Review by lehins — APPROVED** [review]
Besides duplication of code, which you mentioned yourself in Voter instance, this PR is ready to go. :+1:

**Pull request merged**